File: js/creditCheck.js
exports.creditCheck = function (Str) {
  let arr = [];
  Str = Str.toString().split("").reverse();
  for (let i in Str) {
    if (i % 2 !== 0) {
      arr.push(+Str[i] * 2);
    } else {
      arr.push(+Str[i]);
    }
  }
  for (let j in arr) {
    if (arr[j] > 9) {
      let bigger = arr[j].toString().split("");
      arr[j] = +bigger[0] + +bigger[1];
    }
  }
  let res = arr.reduce((curr, pre) => curr + pre, 0);
  return res % 10 == 0 ? "The number is valid!" : "The number is invalid!";
};

File: python/credit_check.py
import functools
def credit_check(Str):
    arr = []
    Str = list(Str)
    for i in range(len(Str)-1, -1, -1):
        num = int(Str[i])
        if i % 2 == 0:
            num *= 2
            if (len(str(num)) > 1):
                splitedNum = list(str(num))
                num = int(splitedNum[0]) + int(splitedNum[1])
        arr.insert(0, num)
    sum = functools.reduce(lambda total, curr : total + curr, arr)
    return "The number is valid!" if sum % 10 == 0 else "The number is invalid!"


